static/src/javascripts/projects/common/modules/navigation/sticky-social-buttons.js

```diff
@@ -0,0 +1,55 @@
+define([
+    'fastdom',
+    'common/utils/$',
+    'common/utils/_',
+    'common/utils/mediator'
+], function (
+    fastdom,
+    $,
+    _,
+    mediator
+) {
+
+    var selectorTopEl = '.social--top',
+        selectorNavSwapOutEl = '.l-header-pre',
+
+        deadzone = 50,
+
+        topEl = _.memoize(function () { return $(selectorTopEl)[0]; }),
+        topElParent = _.memoize(function () { return $(topEl()).parent()[0]; }),
+        navSwapOutEl = _.memoize(function () { return $(selectorNavSwapOutEl)[0]; }),
+
+        isStuck = false;
+
+    function setStickiness() {
+        fastdom.read(function () {
+            var topPos = topElParent().getBoundingClientRect().top;
+
+            if (!isStuck && topPos + deadzone * 2 < 0) {
+                isStuck = stick();
+            } else if (isStuck && topPos + deadzone > 0) {
+                isStuck = unStick();
+            }
+        });
+    }
+
+    function stick() {
```

Comment:
The role of the `isStuck` bool is to avoid redundant DOM reads/writes, eg. having to do `.hasClass(..)`